Tasks/PublishSymbols/PublishSymbolsTask.ps1

```diff
@@ -0,0 +1,44 @@
+[CmdletBinding()]
+param()
+
+Trace-VstsEnteringInvocation $MyInvocation
+try {
+    $symbolServerType = Get-VstsInput -Name 'SymbolServerType' -Require
+
+    if ($SymbolServerType -eq "FileShare")
+    {
+        & "$PSScriptRoot\PublishSymbols.ps1"
+    }
+    elseif ($symbolServerType -eq "TeamServices")
+    {
+        [string]$teamProject = Get-VstsTaskVariable -Name 'System.TeamProject' -Require
+        [string]$buildNumber = Get-VstsTaskVariable -Name 'Build.BuildNumber' -Require
+        [string]$buildId = Get-VstsTaskVariable -Name 'Build.BuildId' -Require
+        [string]$serviceUri = Get-VstsTaskVariable -Name 'System.TeamFoundationCollectionUri' -Require
+
+        [string]$requestName = "$teamProject/$buildNumber/$buildId/$([Guid]::NewGuid())".ToLower()
+        [string]$symbolServiceUri = "$serviceUri" -replace ".visualstudio.com",".artifacts.visualstudio.com"
```

Comment:
if you have a C# rest client for your service then you can use that (lib has functions to load and initialize connection) and that may eliminate url manipulation